### ISSUE TITLE
Backup quizzes where all questions are GGB qtype

### ISSUE THREAD

**Issue by TWINGSISTER** [mebis-lp/moodle-qtype_geogebra#76]
There are two ways of providing tests for general audience on the internet. One is to publish a XML question bank and another is to provide a mbz partial course backup containing just the quiz. The latter is by far more interesting since if you go for the former then you have to build the test yourself. This duplicates efforts from teachers. It would be nice to have a load and go cartridge with a quiz with e.g. ten questions in it.... Starting from this point of view I found that when you try to backup a quiz made up of ALL AND ONLY ggb qtype questions the quiz cannot be selected without checking the question bank backup inclusion that, actually is useless being the GGB content already in the backup.

**Comment by PhMemmel** [comment]
Well...I'm not too experienced with the functionality, but I believe that questions are *always* stored in the question bank. Creating a quiz just means that you add "links" to questions in the question bank to your quiz.
If you add a new question directly to your quiz, moodle internally just adds this question to the question bank first and then links it again to your quiz. With this technical background it is of course necessary to also check the question bank inclusion in a backup. Because that's where the questions are stored.

Apart from that I pretty much believe all other question types have the same issue, so you might want to try out if other question types do not have this behavior. If they do, this is not a `qtype_geogebra` specific issue, but you should discuss this in the moodle forums and eventually raise a ticket in the tracker, if there is none yet.

**Comment by TWINGSISTER** [comment]
You are right, this is a general feature. To have some encapsulation you have to put all the questions for a quiz in a question bank category. Moodle 4.0 is clever enough to add just this category in the backup. The point was to have this plugin to take advantage of the particular  situation for GGB activites and generate thin mbz for GGB quiz distribution.
